test/integration/include/findAll.test.js

```diff
@@ -2054,4 +2054,53 @@ describe(Support.getTestDialectTeaser('Include'), () => {
         });
     });
   });
+
+  it.only('should be able to generate a correct request with inner and outer join', function() {
+    const Customer = this.sequelize.define('customer', {
+      name: DataTypes.STRING
+    });
+
+    const ShippingAddress = this.sequelize.define('shippingAddress', {
+      address: DataTypes.STRING,
+      verified: DataTypes.BOOLEAN
+    });
+
+    const Order = this.sequelize.define('purchaseOrder', {
+      description: DataTypes.TEXT
+    });
+
+    const Shipment = this.sequelize.define('shipment', {
+      trackingNumber: DataTypes.STRING
+    });
+
+    Customer.hasMany(ShippingAddress);
+    ShippingAddress.belongsTo(Customer);
+
+    Customer.hasMany(Order);
+    Order.belongsTo(Customer);
+
+    Shipment.belongsTo(Order);
+    Order.hasOne(Shipment);
+
+    return this.sequelize.sync({ force: true })
+    .then(() => {
+      return Shipment.findOne({
+        include: [{
+          model: Order,
+          required: true,
+          include: [{
+            model: Customer,
+            include: [{
+              model: ShippingAddress,
+              where: { verified: true }
+            }]
+          }]
+        }]
+      })
+      .then(result => {
+        //We just want to check that the query correctly runs
+        expect(result).to.not.be.ok;
```

Comment:
I don't want the assertion to do anything, it's just there to test something at the end to be sure that the query runs and everything's OK. If the query fails, It will not pass in "then".
This assertion as no other purpose than being sure the code goes into "then".